spec.html

```diff
@@ -19462,15 +19476,61 @@ <h1>Runtime Semantics: Evaluation</h1>
 
         <emu-grammar>ImportCall : `import` `(` AssignmentExpression `)`</emu-grammar>
         <emu-alg>
-          1. Let _referencingScriptOrModule_ be GetActiveScriptOrModule().
+          1. Let _referrer_ be GetActiveScriptOrModule().
+          1. If _referrer_ is *null*, set _referrer_ to the current Realm Record.
           1. Let _argRef_ be ? Evaluation of |AssignmentExpression|.
           1. Let _specifier_ be ? GetValue(_argRef_).
           1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
           1. Let _specifierString_ be Completion(ToString(_specifier_)).
           1. IfAbruptRejectPromise(_specifierString_, _promiseCapability_).
-          1. Perform HostImportModuleDynamically(_referencingScriptOrModule_, _specifierString_, _promiseCapability_).
+          1. Let _state_ be a new ModuleLoadState Record { [[Action]]: ~dynamic-import~, [[PromiseCapability]]: _promiseCapability_ }.
+          1. Perform HostLoadImportedModule(_referrer_, _specifierString_, *undefined*, _state_).
           1. Return _promiseCapability_.[[Promise]].
         </emu-alg>
+
+        <emu-clause id="sec-ContinueDynamicImport" type="abstract operation">
+          <h1>
+            ContinueDynamicImport (
+              _state_: a ModuleLoadState Record whose [[Action]] is ~dynamic-import~,
+              _module_: either a normal completion containing a Module Record or a throw completion,
```

Comment:
Perhaps use `_moduleCompletion_` to avoid overloading the `_module_` variable type?